deps: Invalid terraform-plugin-codegen-spec Reference

### tfplugingen-framework CLI version

Current `main` HEAD (03a9e455a82729007a3a799baaa5ed32bc6db33e)

### Provider Code Spec File

```JSON
N/A
```


### Expected Behavior

Installing locally via `go install ./cmd/tfplugingen-framework` to succeed.

### Actual Behavior

```console
$ go install ./cmd/tfplugingen-framework
go: downloading github.com/hashicorp/terraform-plugin-codegen-spec v0.1.1-0.20240104173918-8a979efa31b3
internal/schema/associated_external_type.go:11:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
internal/schema/associated_external_type.go:12:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
internal/datasource/bool_attribute.go:10:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
internal/datasource/convert.go:10:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
internal/provider/bool_attribute.go:10:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
internal/resource/bool_attribute.go:10:2: github.com/hashicorp/terraform-plugin-codegen-spec@v0.1.1-0.20240104173918-8a979efa31b3: invalid version: unknown revision 8a979efa31b3
```

### Additional Information

Looks like the `go.mod` is pointing at a pre-squash merge commit from https://github.com/hashicorp/terraform-plugin-codegen-spec/pull/77.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,7 @@ go 1.21
 require (
 	github.com/google/go-cmp v0.6.0
 	github.com/hashicorp/cli v1.1.6
-	github.com/hashicorp/terraform-plugin-codegen-spec v0.1.1-0.20240104173918-8a979efa31b3
+	github.com/hashicorp/terraform-plugin-codegen-spec v0.1.1-0.20240314092159-611f5a8dcd3c
 	github.com/mattn/go-colorable v0.1.13
 )
 

--- a/internal/cmd/testdata/custom_and_external/all_output/default_pkg_name/resource_example/example_resource_gen.go
+++ b/internal/cmd/testdata/custom_and_external/all_output/default_pkg_name/resource_example/example_resource_gen.go
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
@@ -29,6 +30,7 @@ func ExampleResourceSchema(ctx context.Context) schema.Schema {
 				CustomType: my_bool_type,
 				Computed:   true,
 				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
 					myboolplanmodifier.Modify(),
 				},
 				Validators: []validator.Bool{

--- a/internal/cmd/testdata/custom_and_external/all_output/specified_pkg_name/example_resource_gen.go
+++ b/internal/cmd/testdata/custom_and_external/all_output/specified_pkg_name/example_resource_gen.go
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
@@ -29,6 +30,7 @@ func ExampleResourceSchema(ctx context.Context) schema.Schema {
 				CustomType: my_bool_type,
 				Computed:   true,
 				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
 					myboolplanmodifier.Modify(),
 				},
 				Validators: []validator.Bool{

--- a/internal/cmd/testdata/custom_and_external/ir.json
+++ b/internal/cmd/testdata/custom_and_external/ir.json
@@ -1077,7 +1077,14 @@
               },
               "plan_modifiers": [
                 {
-                  "requires_replace": {}
+                  "custom": {
+                    "imports": [
+                      {
+                        "path": "github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+                      }
+                    ],
+                    "schema_definition": "boolplanmodifier.RequiresReplace()"
+                  }
                 },
                 {
                   "custom": {

--- a/internal/cmd/testdata/custom_and_external/resources_output/example_resource_gen.go
+++ b/internal/cmd/testdata/custom_and_external/resources_output/example_resource_gen.go
@@ -9,6 +9,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/attr"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
 	"github.com/hashicorp/terraform-plugin-framework/types"
@@ -29,6 +30,7 @@ func ExampleResourceSchema(ctx context.Context) schema.Schema {
 				CustomType: my_bool_type,
 				Computed:   true,
 				PlanModifiers: []planmodifier.Bool{
+					boolplanmodifier.RequiresReplace(),
 					myboolplanmodifier.Modify(),
 				},
 				Validators: []validator.Bool{

